common/mixnet-contract/src/events.rs

```diff
@@ -350,3 +349,29 @@ pub fn new_mix_delegators_rewarding_event(
             further_delegations.to_string(),
         )
 }
+
+/// Looks up value of particular attribute in the provided event. If it fails to find it,
+/// the function panics.
+///
+/// # Arguments
+///
+/// * `event`: event to search through.
+/// * `key`: key associated with the particular attribute
+pub fn must_find_attribute(event: &Event, key: &str) -> String {
+    may_find_attribute(event, key).unwrap()
+}
+
+/// Looks up value of particular attribute in the provided event. Returns None if it does not exist.
+///
+/// # Arguments
+///
+/// * `event`: event to search through.
+/// * `key`: key associated with the particular attribute
+pub fn may_find_attribute(event: &Event, key: &str) -> Option<String> {
+    for attr in &event.attributes {
+        if attr.key == key {
+            return Some(attr.value.clone());
+        }
+    }
+    None
+}
```

Comment:
Why not just `may_find_attribute(event, key).ok()`?

Reply:
I think you were looking at an old revision because this file doesn't exist anymore : )

Reply:
Disregard this comment an the other one, I've misread the code